Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -12287,8 +12300,128 @@ def get_server_config():
+
+            def dump_result():
+                """
+                Helper function to dump the profiling results to a file and return the file.
+                """
+                import tempfile
+                # Create a temporary file to store profiling data
+                with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+                    profiler.dump_stats(temp_file.name)
+                    temp_file_path = temp_file.name
+
+                # Read the profiling data from the temporary file
+                with open(temp_file_path, 'rb') as f:
+                    profiling_results = f.read()
+
+                # Delete the temporary file
+                os.remove(temp_file_path)
+
+                # Use Demisto's fileResult to create a file from the profiling stats
+                demisto.results(fileResult('profiling_stats.prof', profiling_results))
+
+            # 5 minutes in nanoseconds
+            default_timeout = 60 * 5 * 1e9
+
+            # The system timeout configuration.
+            timeout_nanoseconds = demisto.callingContext["context"].get("TimeoutDuration") or default_timeout
+            timeout_seconds = timeout_nanoseconds / 1e9
+            failed_on_timeout = False
+            event_set = signal_event.wait(timeout_seconds - 5)
+            if not event_set:
+                failed_on_timeout = True
+            profiler.disable()
+            dump_result()
+            demisto.debug("Profiler finished.")
+            return failed_on_timeout
+
+        def function_runner(func, profiler, signal_event,
+                            results, *args, **kwargs):
+            """
+            A wrapper function that runs the targeted profiled function and captures the results in the results list.
+            :param func: The function to be profiled.
+            :param profiler: The Profile object to use for profiling.
+            :param signal_event: The event to signal when profiling is complete.
+            :param results: A shared object to store the results of the profiled function.
+            :param args: The positional arguments to be passed to the profiled function.
+            :param kwargs: The keyword arguments
+            """
+            profiler.enable()
+            demisto.debug("Profiler started.")
+            try:
+                results["function_results"] = func(*args, **kwargs)
+            finally:
+                # Signal the profiling thread that the command has completed.
+                signal_event.set()
+
+        support_multithreading()
+        results = {}
+        profiler = cProfile.Profile()
+        signal_event = threading.Event()
+        profiler_thread = threading.Thread(target=function_runner, args=(func, profiler, signal_event, results) + args,
+                                           kwargs=kwargs)
+        profiler_thread.start()
+
+        failed_on_timeout = profiler_function(signal_event, profiler)
+        if failed_on_timeout:
```

Comment:
do we need an eception? I think the original error will propogate through this. Lets just log that this happened in demisto.error (and double check that this works)

Reply:
I will try to remove this and test it out.